pin/pin.go

```diff
@@ -132,72 +130,18 @@ func (p *pinner) Unpin(ctx context.Context, k key.Key, recursive bool) error {
 	if p.recursePin.HasKey(k) {
 		if recursive {
 			p.recursePin.RemoveBlock(k)
-			node, err := p.dserv.Get(ctx, k)
-			if err != nil {
-				return err
-			}
-
-			return p.unpinLinks(ctx, node)
+			return nil
 		} else {
 			return fmt.Errorf("%s is pinned recursively", k)
 		}
 	} else if p.directPin.HasKey(k) {
 		p.directPin.RemoveBlock(k)
 		return nil
-	} else if p.indirPin.HasKey(k) {
-		return fmt.Errorf("%s is pinned indirectly. indirect pins cannot be removed directly", k)
```

Comment:
we should keep this error. it lets the user know _why_ something won't be gc-ed. suppose

- i notice i have a huge file added.
- i try to unpin it. it fails with an error that **it's not pinned**.
- i GC.
- it's still there.
- [wat](https://i.imgur.com/IppKJ.jpg).